salt/states/pkg.py

```diff
@@ -743,7 +743,15 @@ def _find_install_targets(
     warnings = []
     failed_verify = False
     for package_name, version_string in desired.items():
-        cver = cur_pkgs.get(package_name, [])
+
+        # FreeBSD pkg supports `openjdk` and `java/openjdk7` package names
+        origin = bool(re.search('/', package_name))
+
+        if __grains__['os'] == 'FreeBSD' and origin:
+            cver = [k for k, v in cur_pkgs.items() if v['origin'] == package_name]
```

Comment:
There are often hundreds or thousands of packages installed on a system. If someone is attempting to install many of these packages that have a `/` in the name then iterating over all the packages every time (O(n*m) might not be as performant as one would desire; although it could be an acceptable since there are probably not millions or hundreds of thousands of packages.

Reply:
It appears there is a orgins dict stored in the context (https://github.com/saltstack/salt/blob/9233e1cc3b6b072a61b445d285ba856fc642ef3b/salt/modules/pkgng.py#L171); perhaps that could be utilized.